library/std/src/env.rs

```diff
@@ -112,6 +114,19 @@ pub struct VarsOs {
 /// }
 /// ```
 ///
+/// ```
+/// use std::env;
+///
+/// let before_set = env::vars().count();
+/// env::set_var("TEST_ADD_REMOVE_ENV_VAR", "42");
+/// let before_remove = env::vars().count();
+/// env::remove_var("TEST_ADD_REMOVE_ENV_VAR");
+/// let after_remove = env::vars().count();
```

Comment:
But this only creates iterators and immediately calculates their counts, it does not exercise the snapshotting in any form.